firefox-ios/Client/TabManagement/Legacy/LegacyTabManager.swift

```diff
@@ -631,6 +631,16 @@ class LegacyTabManager: NSObject, FeatureFlaggable, TabManager, TabEventHandler
         storeChanges()
     }
 
+    @MainActor
+    func removeTabByUrl(_ url: URL) async -> Bool {
```

Comment:
```suggestion
    func removeTabs(with urls: [URL]) async -> Bool {
```

Thought: How would you feel about changing this signature to take multiple URLs (and then changing `ApplicationHelper.closeTab` to `closeTabs`?)

That way, we can turn `urls` into a `Set`, filter `normalTabs` to find the tabs for all the URLs in the set in just one pass, then remove each one. It would make this an `O(tabs.count)` operation, rather than `O(tabs.length * urls.count)`.

Something like:

```swift
func removeTabs(with urls: [URL]) async {
    let urls = Set(urls)
    let tabsToRemove = normalTabs.filter { tab in
        guard let url = tab.url else { return false }
        return urls.contains(url)
    }
    for tab in tabs {
        await withCheckedContinuation { continuation in
            removeTab(tab) { continuation.resume() }
        }
    }
}
```

(As a bonus, we could also optimize `removeTab(_:completion:)` to update the selected index just once, but one step at a time!)